sds/src/path.rs

```diff
@@ -52,8 +52,20 @@ impl<'a> Path<'a> {
         true
     }
 
+    fn size(&self) -> usize {
+        self.segments
+            .iter()
+            .map(|segment| {
+                if let PathSegment::Field(field) = segment {
+                    return field.len();
+                }
+                0
+            })
+            .sum()
+    }
+
     pub fn sanitize(&self) -> String {
-        let mut sanitized_path = "".to_string();
+        let mut sanitized_path = String::with_capacity(self.size() * 2);
```

Comment:
`*2` seems a bit much. How did the benchmarks look when you changed just this 1 line? We could probably get away with a smaller number, maybe 1.5 or so. This doesn't necessarily have to be the upper bound, if the capacity is set too small it will still grow automatically as needed. It would be interesting to know how much time is spent in this allocation vs the rest of the sanitization code to be able to fine-tune this better. It could end up being worth doing a quick scan to get the exact size need (especially if that scan is combined with the check to skip sanitization all-together)

Reply:
I've used 1.5 times, thanks for the comment :)